openwpm/config.py

```diff
@@ -0,0 +1,263 @@
+import os
+from dataclasses import dataclass, field
+from typing import List, Tuple
+
+from dataclasses_json import dataclass_json
+
+from .errors import ConfigError
+
+BOOL_TYPE_VALIDATION_LIST = [True, False]
+DISPLAY_MODE_VALIDATION_LIST = ["native", "headless", "xvfb"]
+SUPPORTED_BROWSER_LIST = [
+    "firefox"
+]  # Using List instead of a str type to future proof the logic as OpenWPM may add support for more browsers in future
+TP_COOKIES_OPTIONALS_LIST = ["always", "never", "from_visited"]
+DB_EXTENSION_TYPE_LIST = [".db", ".sqlite"]
+LOG_EXTENSION_TYPE_LIST = [".log"]
+CONFIG_ERROR_STRING = (
+    "Found {value} as value for {parameter_name} in BrowserParams. "
+    "Supported values are {value_list}. Please look at "
+    "docs/Configuration.md#browser-configuration-options for more information"
+)
+EXTENSION_ERROR_STRING = (
+    "Found {extension} extension for {parameter_name} in ManagerParams "
+    "supported extensions are {value_list}. Please look at "
+    "docs/Configuration.md#platform-configuration-options for more information"
+)
+GENERAL_ERROR_STRING = (
+    "Found invalid value `{value}` for {parameter_name} in {params_type}. "
+    "Please look at docs/Configuration.md for more information"
+)
+OUTPUT_FORMAT_LIST = ["local", "s3"]
+
+ALL_RESOURCE_TYPES = {
+    "beacon",
+    "csp_report",
+    "font",
+    "image",
+    "imageset",
+    "main_frame",
+    "media",
+    "object",
+    "object_subrequest",
+    "ping",
+    "script",
+    "stylesheet",
+    "sub_frame",
+    "web_manifest",
+    "websocket",
+    "xbl",
+    "xml_dtd",
+    "xmlhttprequest",
+    "xslt",
+    "other",
+}
+
+
+@dataclass_json
+@dataclass
+class BrowserParams:
+    extension_enabled: bool = True
+    cookie_instrument: bool = True
+    js_instrument: bool = False
+    js_instrument_settings: list = field(
+        default_factory=lambda: ["collection_fingerprinting"]
+    )
+    http_instrument: bool = False
+    navigation_instrument: bool = False
+    save_content: bool = False
```

Comment:
```suggestion
    save_content: Union[bool, str] = False
```
Since save_content can also be a string we should say so in the type.
You'll need to `from collections import Union`